prompt-service/src/unstract/prompt_service/services/answer_prompt.py

```diff
@@ -255,7 +255,10 @@ def handle_json(
 
             # Heuristic: if wrapping only added '[' and ']', len(b) - len(a) == 2 → original was valid, use 'a'
             # Otherwise, fallback to 'b' which likely fixed multiple items or invalid top-level structure
-            parsed_data = a if len(json.dumps(b)) - len(json.dumps(a)) == 2 else b
+            dump_a = json.dumps(a, ensure_ascii=False)
+            dump_b = json.dumps(b, ensure_ascii=False)
+            ARRAY_WRAP_DELTA = 2  # '[' and ']'
+            parsed_data = a if len(dump_b) - len(dump_a) == ARRAY_WRAP_DELTA else b
```

Comment:
@Deepak-Kesavan will `len(dump_b) - len(dump_a)` always be positive?
Can repair JSON remove any characters by any chance - should we make it this instead?
```suggestion
            parsed_data = a if abs(len(dump_b) - len(dump_a)) == ARRAY_WRAP_DELTA else b
```

Reply:
No @chandrasekharan-zipstack . It will always be positive. We won't be needing absolute func here.